spec/requests/portfolios_spec.rb

```diff
@@ -295,26 +295,18 @@
       end
     end
 
-    RSpec.shared_context "sharing_objects" do
-      let(:group_uuids) { %w[1 2 3] }
-      let(:permissions) { %w[catalog:portfolios:read] }
-      let(:app_name) { "catalog" }
-      let(:http_status) { '204' }
-      let(:dummy) { double("Insights::API::Common::RBAC::ShareResource", :process => self) }
-      let(:attributes) { {:group_uuids => group_uuids, :permissions => permissions} }
-    end
 
     shared_examples_for "#shared_test" do
       it "portfolio" do
         with_modified_env :APP_NAME => app_name do
-          options = {:app_name      => app_name,
-                     :resource_ids  => [portfolio.id.to_s],
-                     :resource_name => 'portfolios',
-                     :permissions   => permissions,
-                     :group_uuids   => group_uuids}
-          allow(Insights::API::Common::RBAC::ShareResource).to receive(:new).with(options).and_return(dummy)
-          post "#{api}/portfolios/#{portfolio.id}/share", :params => attributes, :headers => default_headers
+          allow(rs_class).to receive(:call).with(RBACApiClient::GroupApi).and_yield(api_instance)
+          allow(Insights::API::Common::RBAC::Service).to receive(:paginate).with(api_instance, :list_groups, {}).and_return(groups)
+          post "#{api}/portfolios/#{shared_portfolio.id}/share", :params => attributes, :headers => default_headers
           expect(response).to have_http_status(http_status)
+          if http_status == 204
+            shared_portfolio.reload
+            expect(shared_portfolio.access_control_entries.count).to eq(groups.count)
+          end
```

Comment:
This is kinda weird to me, it's just forcing a spec to conform to different outcomes, when the point of a shared example is to expect the same behavior. I'm all for DRY-ing up specs, but in this instance, it seems like the 204 is the odd-man out and that should be a separate test, and we leave this shared example for the "bad permissions" set of specs.